src/legacy/core_plugins/vis_default_editor/public/components/controls/field.tsx

```diff
@@ -104,7 +104,8 @@ function FieldParamEditor({
   return (
     <EuiFormRow
       label={customLabel || label}
-      isInvalid={showValidation ? !isValid : false}
+      // we show an error message right away if there is no compatible fields
+      isInvalid={showValidation || !indexedFields.length ? !isValid : false}
```

Comment:
This means the row will be invalid if `showValidation` is true ?
Actually this check is really hard to understand.
What if you move the check of `indexedFields.length` :
```
  const isValid = !!value && !errors.length && !isDirty && !!indexedFields.length;

<EuiFormRow
    isInvalid={showValidation && !isValid }
```

Reply:
The main idea here is to show the error message before a user touches the form since they anyway won't be able to make the form valid when there are no compatible fields.

>const isValid = !!value && !errors.length && !isDirty && !!indexedFields.length;

`errors` will contain a message when there are no indexedFields.